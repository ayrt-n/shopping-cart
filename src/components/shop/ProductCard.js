import React from 'react';
import AddToCartForm from './AddToCartForm';

function ProductCard({ name, image, price, handleSubmit }) {
  const imageSrc = require(`./images/${image}`);

  const addToCart = (quantity) => {
    handleSubmit(
      {
        name,
        image,
        price,
        quantity,
      }
    )
  }

  return (
    <li className='Shop-grid-item'>
      <img src={imageSrc} alt={image} />
      <h2>{name}</h2>
      <p className='Shop-price'>${price}</p>
      <AddToCartForm handleSubmit={addToCart} />
    </li>
  );
}

export default ProductCard;
