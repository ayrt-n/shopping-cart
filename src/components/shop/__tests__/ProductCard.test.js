import React, { useEffect } from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import ProductCard from '../ProductCard';

// Setup mocks
jest.mock('../AddToCartForm', () => ({ handleSubmit }) => (
  <button onClick={() => handleSubmit(1)}>Mock Form</button>
));

// ProductCard Tests
describe('ProductCard component', () => {
  it('correctly renders product card', () => {
    const product = { name: 'Test', image: 'test.jpg', price: 1 };

    const { container } = render(
      <ProductCard
        name={product.name}
        image={product.image}
        price={product.price}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('calls handleSubmit with correct arguments when called from AddToCartForm', () => {
    const addToCartMock = jest.fn();
    const product = { name: 'Test', image: 'test.jpg', price: 1 };

    render(
      <ProductCard
        name={product.name}
        image={product.image}
        price={product.price}
        handleSubmit={addToCartMock}
      />
    );

    const addToCartButton = screen.getByRole('button', {name: /mock form/i});
    userEvent.click(addToCartButton);

    expect(addToCartMock).toHaveBeenCalledTimes(1)
    expect(addToCartMock).toHaveBeenCalledWith({
      ...product,
      quantity: 1
    })
  });
});
