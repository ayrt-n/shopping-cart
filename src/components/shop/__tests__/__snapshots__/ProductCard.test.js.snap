// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component correctly renders product card 1`] = `
<div>
  <li
    class="Shop-grid-item"
  >
    <img
      alt="test.jpg"
      src="test-file-stub"
    />
    <h2>
      Test
    </h2>
    <p
      class="Shop-price"
    >
      $
      1
    </p>
    <button>
      Mock Form
    </button>
  </li>
</div>
`;
